Sources/Peasy/Internal/Socket.swift

```diff
@@ -30,9 +30,26 @@ final class Socket {
 		address.sin6_port = UInt16(port).bigEndian
 		address.sin6_addr = .localhost
 		let size = socklen_t(MemoryLayout<sockaddr_in6>.size)
-		let success = withUnsafePointer(to: &address) { $0.withMemoryRebound(to: sockaddr.self, capacity: Int(size)) { Darwin.bind(tag, $0, size) >= 0 } }
+		let success = withUnsafePointer(to: &address) { address -> Bool in
+			return address.withMemoryRebound(to: sockaddr.self, capacity: Int(size)) {
+				Darwin.bind(tag, $0, size) >= 0
+			}
+		}
+
+		var usedAddressSize = socklen_t(MemoryLayout<sockaddr_in6>.size)
+		var usedAddress = sockaddr_in6()
+		_ = withUnsafeMutablePointer(to: &usedAddress) { usedAddress in
+			_ = withUnsafeMutablePointer(to: &usedAddressSize) { usedAddressSize in
+				usedAddress.withMemoryRebound(to: sockaddr.self, capacity: Int(size)) {
+					Darwin.getsockname(tag, $0, usedAddressSize)
+				}
+			}
+		}
```

Comment:
Man these Darwin APIs really make me appreciate how far API design has come on 😂

Reply:
I was able to get rid of one level, as `usedAddressSize` can be passed via `&usedAddressSize`.